ADR/0035-apps-continuous-chaos-testing.md

```diff
@@ -0,0 +1,44 @@
+Establish and define your steady state and metrics - understand the behavior and performance under stable conditions and define the metrics that will be used to evaluate the application’s behavior. Then decide on acceptable outcomes before injecting chaos.
+Analyze the statuses and metrics of all components during the chaos test runs.
+Improve the areas that are not resilient and performant by comparing the key metrics and Service Level Objectives (SLOs) to the stable conditions before the chaos. For example: evaluating the API server latency or application uptime to see if the key performance indicators and service level indicators are still within acceptable limits.
+
+![Architecture diagram of SLA based Chaos test scenario](../diagrams/ADR-0035/chaos-sla.png "Architecture diagram of SLA based Chaos test scenario")
+
+
+### Glossary
+
+- krkn: Chaos testing framework: <https://github.com/krkn-chaos/krkn>
+
+## Decision
+
+The Knoflux user has the ability to execute chaos tests such as Krkn as a part of the IntegrationTestScenarios by utilizing ephemeral clusters ([provisioning-ephemeral-openshift-clusters](https://github.com/konflux-ci/architecture/pull/172)) instead of ephemeral namespaces for enhanced isolation and a more production-like testing environment. Furthermore, they can gather Prometheus metrics from Thanos-querier-openshift-monitoring or Prometheus-k8s-openshift-monitoring endpoints.
+
+## Consequences
+
+* The service/user account associated with the ephemeral environment will have cluster admin level privileges to execute CRUD operations (configure RBAC permissions, prometheus instances) within the ephemeral environment. At a minimum, the user should be granted administrator level privilege for a namespace on the ephemeral cluster.
+* Using the service account token, it will be feasible to authenticate and query Prometheus metrics from either `Thanos-querier-openshift-monitoring` or `Prometheus-k8s-openshift-monitoring` endpoints. The privilege could be granted by assigning a `cluster-monitoring-view` role to the associated account, which would permit the user to query prometheus metrics across all namespaces in the cluster.
+* The user account shall have the capability to query the `Thanos-querier-openshift-monitoring` or `Prometheus-k8s-openshift-monitoring` route from the `openshift-monitoring` namespace. The account should have at least `view` level access to the `openshift-monitoring` namespace.
+* An additional openshfit feature that is recommended, though not mandatory, is the provision of [monitoring for user-defined projects](https://docs.openshift.com/container-platform/4.15/observability/monitoring/enabling-monitoring-for-user-defined-projects.html#accessing-metrics-from-outside-cluster_enabling-monitoring-for-user-defined-projects). `Monitoring-rules-edit` and `monitoring-edit` would be associated with the account to perform CURD operations related to `PrometheusRule`, `ServiceMonitor` and `PodMonitor` resources.
```

Comment:
This point seems more like an implementation detail rather than a consequence. I think it fits better under the decision section as an option to consider.

Reply:
Hello @amisstea , thank you for the input, will move it to the decision section.